node/lib/util/rebase.js

```diff
@@ -42,6 +42,18 @@ const SubmoduleUtil = require("./submodule_util");
 const UserError     = require("./user_error");
 
 /**
+ * Put the head of the specified `repo` on the specified `commitSha`.
+ */
+const setHead = co.wrap(function *(repo, commitSha) {
+    // TODO: use a more "gentle" strategy that won't stomp on untracked files.
+    // The checkout strategy I was using doesn't handle all situations well,
```

Comment:
I feel like comments ought to explain why the code is the way it is, not how it used to be.

So, "checkout won't work because it doesn't affect nonconflicting staged changes" would work.